pkg/schemadsl/compiler/compiler.go

```diff
@@ -19,37 +21,73 @@ type InputSchema struct {
 	SchemaString string
 }
 
+// SchemaDefinition represents an object or caveat definition in a schema.
+type SchemaDefinition interface {
+	proto.Message
+
+	GetName() string
+}
+
+// CompiledSchema is the result of compiling a schema when there are no errors.
+type CompiledSchema struct {
+	// ObjectDefinitions holds the object definitions in the schema.
+	ObjectDefinitions []*core.NamespaceDefinition
+
+	// CaveatDefinitions holds the caveat definitions in the schema.
+	CaveatDefinitions []*core.CaveatDefinition
+
+	// OrderedDefinitions holds the object and caveat definitions in the schema, in the
+	// order in which they were found.
+	OrderedDefinitions []SchemaDefinition
+}
+
 // Compile compilers the input schema(s) into a set of namespace definition protos.
-func Compile(schemas []InputSchema, objectTypePrefix *string) ([]*core.NamespaceDefinition, error) {
+func Compile(schemas []InputSchema, objectTypePrefix *string) (*CompiledSchema, error) {
```

Comment:
Honestly? I don't think we use the slice... this might just be something I intended to support initially with having multiple schema blocks and then never did.

I'm going to change to just take in a single schema